File: Golang/Easy/3000_Maximum_Area_of_Longest_Diagonal_Rectangle/solution.go
package maximumareaoflongestdiagonalrectangle

import "math"

func areaOfMaxDiagonal(dimensions [][]int) int {
	maxD := float64(0)
	maxArea := 0

	for i := range dimensions {
		l := dimensions[i][0]
		b := dimensions[i][1]
		currentD := math.Sqrt(float64(l*l + b*b))
		if currentD > maxD {
			maxD = currentD
			maxArea = l * b
		} else if currentD == maxD {
			maxArea = max(maxArea, l*b)
		}
	}
	return maxArea
}

File: Python/Easy/3000_Maximum_Area_of_Longest_Diagonal_Rectangle/solution.py
import math

class Solution:
    def areaOfMaxDiagonal(self, dimensions: list[list[int]]) -> int:
        maxD = 0
        maxArea = 0

        for d in range(len(dimensions)):
            l, b = dimensions[d]
            currentD = math.sqrt(l*l + b*b)
            if currentD > maxD:
                maxD = currentD
                maxArea = l*b
            elif currentD == maxD:
                maxArea = max(maxArea, l*b)
        return maxArea